tools/codegen/cmd/mixgeninventory/main.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"istio.io/mixer/tools/codegen/pkg/inventory"
+)
+
+func withArgs(args []string, errorf func(format string, a ...interface{})) {
+	var mappings []string
+
+	rootCmd := cobra.Command{
+		Use:   "mixgeninventory",
+		Short: "Generates mixer adapter inventory source code",
+		Long: "Generates mixer adapter inventory source code from an input map of adapter packages to include in the inventory.\n" +
+			"Example: mixgeninventory -p prometheus:istio.io/mixer/adapter/prometheus -p stdio:istio.io/mixer/adapter/stdio",
+		Run: func(cmd *cobra.Command, args []string) {
+			packageMap := make(map[string]string)
+			for _, maps := range mappings {
+				m := strings.Split(maps, ":")
+				packageMap[strings.TrimSpace(m[0])] = strings.TrimSpace(m[1])
+			}
+			if err := inventory.Generate(packageMap, os.Stdout); err != nil {
```

Comment:
I would prefer -o options to make the tool standalone that takes input and produces output. However, I don't care for this particular case, since the stand alone tool that generates inventory has no use outside the mixer's build pipeline. Up to you.